java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptionState.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ */
+package com.joyent.manta.client.multipart;
+
+import com.joyent.manta.client.crypto.EncryptionContext;
+
+import java.io.OutputStream;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Package private class that contains the state of the encryption streaming
+ * ciphers used to encrypt multipart uploads.
+ *
+ * @author <a href="https://github.com/cburroughs/">Chris Burroughs</a>
+ * @since 3.0.0
+ */
+class EncryptionState {
+    /**
+     * Encryption cipher state object.
+     */
+    private final EncryptionContext encryptionContext;
+
+    /**
+     * Lock used coordinate concurrent operations.
+     */
+    private final ReentrantLock lock;
+
+    /**
+     * The number of the last part processed.
+     */
+    private int lastPartNumber = -1;
+
+    /**
+     * The multipart stream that allows for attaching and detaching streams.
+     */
+    private MultipartOutputStream multipartStream = null;
+
+    /**
+     * The encrypting stream.
+     */
+    private OutputStream cipherStream = null;
+
+    /**
+     * Creates a new multipart encryption state object.
```

Comment:
Should probably call out that the constructor does *not* setup the streams for use.